### PR TITLE
Updates to support Secrets Manager instead of Certificate Manager

### PR DESCRIPTION
**Pull request by dprosper**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -73,4 +73,5 @@ script:
       --env TRAVIS \
       --env TERRAFORM_VERSION \
       --env SHOW_VERSIONS=true \
+      --env SM_INSTANCE_NAME=$SM_INSTANCE_NAME \
```
Comment (l2fprod) [inline .travis.yml:76]: @dprosper  you don't need `=$SM_INSTANCE_NAME`, just `--env SM_INSTANCE_NAME \` will create the env variable and takes its value

File: vpc-cockroachdb-mzr/variables.tf [hunk 2]
```diff
@@ -70,15 +70,28 @@ variable "vpc_database_image_name" {
...
\ No newline at end of file
+  default     = "cockroach-v20.2.4.linux-amd64"
+}
+
+variable "create_secrets_manager_instance" {
+  description = "Indicates whether or not to create a Secrets Manager instance, values can be true or false, default to false."
+  default     = false
+}
+
+variable "secrets_manager_instance_name" {
```
Comment (l2fprod) [inline vpc-cockroachdb-mzr/variables.tf:91]: @dprosper 

I guess this should have a default empty value as if create == true you will not use the name but without a default, you would still require it

File: vpc-cockroachdb-mzr/variables.tf [hunk 3]
```diff
@@ -70,15 +70,28 @@ variable "vpc_database_image_name" {
...
+  description = "The name of an existing Secrets Manager instance, used when create_secrets_manager_instance is set to false."
+}
+
+variable "secrets_manager_group_name" {
```
Comment (l2fprod) [inline vpc-cockroachdb-mzr/variables.tf:95]: same for the group name. If you create a new instance, the name could have a good default.

**Review by l2fprod — CHANGES_REQUESTED** [review]

File: vpc-cockroachdb-mzr/variables.tf (hunk 3 above)
Comment (dprosper) [reply vpc-cockroachdb-mzr/variables.tf:95]: added

File: vpc-cockroachdb-mzr/variables.tf (hunk 2 above)
Comment (dprosper) [reply vpc-cockroachdb-mzr/variables.tf:91]: I actually no longer need the name, I am making it based on the prefix

File: vpc-cockroachdb-mzr/cloud_services.tf [hunk 4]
```diff
@@ -26,10 +26,19 @@ resource "ibm_iam_authorization_policy" "policy" {
...
   resource_group_id = data.ibm_resource_group.group.id
-}
\ No newline at end of file
+}
+
+data "ibm_resource_instance" "sm_certs" {
+  count             = tobool(var.create_secrets_manager_instance) == true ? 0 : 1
+  name              = var.secrets_manager_instance_name
+  location          = var.vpc_region
+  resource_group_id = data.ibm_resource_group.group.id
```
[the remark this answers is not in the document]
Comment (dprosper) [reply vpc-cockroachdb-mzr/cloud_services.tf:42]: prompting for the guid now instead and not using the data source

File: .travis.yml (hunk 1 above)
Comment (dprosper) [reply .travis.yml:76]: can you provide the guid instead --- made the following update
image

**Pull request merged**